Prose/ProseLib/Sources/MainWindowFeature/MainScreen.swift

```diff
@@ -40,92 +54,91 @@ public struct MainScreen: View {
 
 // MARK: Reducer
 
-public let mainWindowReducer: Reducer<
+public let mainWindowReducer = Reducer<
     MainScreenState,
     MainScreenAction,
     MainScreenEnvironment
-> = sidebarReducer._pullback(
-    state: \MainScreenState.sidebar,
-    action: CasePath(MainScreenAction.sidebar),
-    environment: \MainScreenEnvironment.sidebar
-)
+>.combine([
+    sidebarReducer.pullback(
+        state: \.sidebar,
+        action: CasePath(MainScreenAction.sidebar),
+        environment: { _ in }
+    ),
+    unreadReducer._pullback(
+        state: (\MainScreenState.route).case(CasePath(MainScreenState.Route.unreadStack)),
+        action: CasePath(MainScreenAction.unreadStack),
+        environment: { _ in .stub }
+    ),
+    conversationReducer._pullback(
+        state: (\MainScreenState.route).case(CasePath(MainScreenState.Route.chat)),
+        action: CasePath(MainScreenAction.chat),
+        environment: { _ in .stub }
+    ),
+    Reducer { state, action, _ in
+        switch action {
+        case .sidebar(.selection(.unreadStack)):
+            state.route = .unreadStack(.init())
+            return .none
+
+        case .sidebar(.selection(.replies)):
+            state.route = .replies
+            return .none
+
+        case .sidebar(.selection(.directMessages)):
+            state.route = .directMessages
+            return .none
+
+        case .sidebar(.selection(.peopleAndGroups)):
+            state.route = .peopleAndGroups
+            return .none
+
+        case let .sidebar(.selection(.chat(jid))):
+            state.route = .chat(.init(chatId: .person(id: jid), recipient: "Huh?"))
```

Comment:
What is this "Huh?"?

`recipient` is the display name of the recipient, currently used in the "… is typing" indicator. I agree it's not perfect, but because everything is not well connected with ProseCore yet, I had to make it a simple `String` like this…

Reply:
Ok, then maybe add a TODO comment?

Reply:
The feature is yet to be done. If you build it according to the design is there any other way than to change this?

Reply:
No no, that's why I said add a TODO comment, so we don't forget it when implementing the feature 😉 But I'm fine if we don't add such a comment 😉

Reply:
I generally don't like TODO message in code. Either fix it or create a ticket. In this case we already have a ticket.